### PR TITLE
refactor: clean up dialog-manager and fixed-dimensions

### PR DESCRIPTION
**Pull request by HendrikThePendric**
Implements [this suggestion](https://github.com/dhis2/event-reports-app/pull/956#discussion_r810831308)

---

### Key features

1. Cleans up the conditional rendering in the `DialogManager` component, which was a bit more complex than it needed to be.
2. Cleans up the code in `FixedDimensions` because it "Looks like it was expecting to handle PERIOD in the future, but now it probably will only ever handle OU, since there is a separate PeriodDimension component." (I've quoted Jennifer here)

---

### CODE REVIEW

File: src/components/Dialogs/DialogManager.js [hunk 1]
```diff
@@ -32,22 +23,24 @@ const DialogManager = () => {
...
+    switch (dimension.dimensionType) {
+        case DIMENSION_TYPE_CATEGORY:
+        case DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET:
+        case DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET: {
+            return <DynamicDimension dimension={dimension} onClose={onClose} />
+        }
+        case DIMENSION_TYPE_PERIOD: {
+            return <PeriodDimension dimension={dimension} onClose={onClose} />
+        }
         case DIMENSION_ID_ORGUNIT: {
```
Comment (jenniferarnesen) [inline src/components/Dialogs/DialogManager.js:39]: Use DIMENSION_TYPE_OU

File: src/components/Dialogs/DialogManager.js [hunk 2]
```diff
@@ -32,22 +23,24 @@ const DialogManager = () => {
...
-        return <DynamicDimension dimension={dimension} onClose={onClose} />
-    }
-    if (dimension?.dimensionType === DIMENSION_TYPE_PERIOD) {
-        return <PeriodDimension dimension={dimension} onClose={onClose} />
+    if (!dimension?.id) {
+        return null
     }
-    switch (dimension?.id) {
-        case DIMENSION_TYPE_PROGRAM_STATUS:
-        case DIMENSION_TYPE_EVENT_STATUS:
```
Comment (martinkrulltott) [inline src/components/Dialogs/DialogManager.js:43]: the handling of `case DIMENSION_TYPE_PROGRAM_STATUS:` and `case DIMENSION_TYPE_EVENT_STATUS:` is missing

File: src/components/Dialogs/FixedDimension.js [hunk 3]
```diff
@@ -54,34 +54,27 @@ const FixedDimension = ({
...
-                const forParentGraphMap = {}
-
-                items.forEach((ou) => {
-                    const id = ouIdHelper.removePrefix(ou.id)
-                    forMetadata[id] = {
-                        id,
-                        name: ou.name || ou.displayName,
-                        displayName: ou.displayName,
-                    }
+        const forMetadata = {}
```
Comment (jenniferarnesen) [inline src/components/Dialogs/FixedDimension.js:57]: Rename file/component to OuDimension

Comment (jenniferarnesen) [reply src/components/Dialogs/FixedDimension.js:57]: Or OrgUnitDimension

Comment (jenniferarnesen) [reply src/components/Dialogs/FixedDimension.js:57]: Or OrgUnitDimension

**Comment by HendrikThePendric** [comment]
Closing this PR for now. It was meant to be a quick refactor, but it became apparent that it would be better to do this more thoroughly. Since we are currently in a crunch to get the MVP ready, now is not the time for this.

**Pull request closed without merging**